conans/test/functional/build_requires/build_requires_test.py

```diff
@@ -54,15 +54,18 @@ class BuildRequiresTest(unittest.TestCase):
 
     def test_consumer(self):
         # https://github.com/conan-io/conan/issues/5425
+        catch_ref = ConanFileReference.loads("catch/0.1@user/testing")
+        libA_ref = ConanFileReference.loads("LibA/0.1@user/testing")
+
         t = TestClient()
-        t.save({"conanfile.py": str(TestConanFile("catch", "0.1", info=True))})
+        t.save({"conanfile.py":
+                    GenConanfile().with_package_info(cpp_info={"libs": ["mylibcatch0.1lib"]},
+                                                     env_info={"MYENV": ["myenvcatch0.1env"]})})
         t.run("create . catch/0.1@user/testing")
-        t.save({"conanfile.py": str(TestConanFile("LibA", "0.1",
-                                                  private_requires=["catch/0.1@user/testing"]))})
+        t.save({"conanfile.py": GenConanfile().with_requirement(catch_ref, private=True)})
         t.run("create . LibA/0.1@user/testing")
-        t.save({"conanfile.py": str(TestConanFile("LibC", "0.1",
-                                                  requires=["LibA/0.1@user/testing"],
-                                                  build_requires=["catch/0.1@user/testing"]))})
+        t.save({"conanfile.py": GenConanfile().with_requirement(libA_ref)
```

Comment:
I don't think it is very important, but these tests have changed from declaring class-attribute ``requires=xxx`` to requires using the ``requirements()`` method.

Is there any specific reason for this change? was it accidental or on purpose?

Reply:
First iteration: everything moved to `requirements()` because it was the only way allowed... but one test failed (https://github.com/conan-io/conan/pull/5654). So, it was accidental and after it only the failing test was moved back to `require=xxx`.

In my mind, it is the same to declare requirements one way or the other, and I think that Conan should behave the same. I would prioritize that PR, but if you feel like there is any difference, then I can revisit this PR and move back some `requirements()` to `requires`.

Reply:
Yes please, recover them. We agreed on that on the daily, to recover all the changed "requires" but keep the detected error to open a new PR because it looked like a bug. My bad to not review it properly before merging this.

Reply:
On my way!